packages/cache/src/cache.ts

```diff
@@ -182,17 +172,31 @@ export async function saveCache(
       await listTar(archivePath, compressionMethod)
     }
 
-    const fileSizeLimit = 10 * 1024 * 1024 * 1024 // 10GB per repo limit
     const archiveFileSize = utils.getArchiveFileSizeInBytes(archivePath)
     core.debug(`File Size: ${archiveFileSize}`)
-    if (archiveFileSize > fileSizeLimit) {
-      throw new Error(
-        `Cache size of ~${Math.round(
-          archiveFileSize / (1024 * 1024)
-        )} MB (${archiveFileSize} B) is over the 10GB limit, not saving cache.`
+
+    const cacheSize = archiveFileSize
+    core.debug('Reserving Cache')
+    let reserveCacheResponse = await cacheHttpClient.reserveCache(key, paths, {
+      compressionMethod,
+      cacheSize
+    })
+
+    if(reserveCacheResponse?.statusCode === 400){
+      throw new ReserveCacheError(
+        `Cache size of ~${Math.round(archiveFileSize / (1024 * 1024))} MB (${archiveFileSize} B) is over the data cap limit, not saving cache.`
```

Comment:
This is wrong and will cause any 400 status code to show this error message to user.